wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/UserClientCell.swift

```diff
@@ -23,142 +23,174 @@ import WireDataModel
+    let proteusIdLabel = DynamicFontLabel(style: .caption1,
+                                            color: LabelColors.textCellSubtitle)
+
+    private let statusStackView = UIStackView()
+    private let contentWrapView = UIView()
+    private let contentStackView = UIStackView()
     private let accessoryIconView = UIImageView()
-    private let verifiedIconView = UIImageView()
 
-    private var contentStackView: UIStackView!
-    private var titleStackView: UIStackView!
-    private var iconStackView: UIStackView!
-
-    private let boldFingerprintFont: UIFont = .smallSemiboldFont
-    private let fingerprintFont: UIFont = .smallFont
-
-    private weak var client: UserClientType?
-
-    override func setUp() {
-        super.setUp()
+    var showLabel: Bool = false {
+        didSet {
+            updateLabel()
+        }
+    }
 
-        accessibilityIdentifier = "device_cell"
+    var viewModel: ClientTableViewCellModel? {
+        didSet {
+            nameLabel.text = viewModel?.title
+            proteusIdLabel.text = viewModel?.proteusLabelText
+            mlsThumbprintLabel.text = viewModel?.mlsThumbprintLabelText
+            statusStackView.removeArrangedSubviews()
+            if let e2eIdentityStatusImage = viewModel?.e2eIdentityStatus?.uiImage {
+                statusStackView.addArrangedSubview(UIImageView(image: e2eIdentityStatusImage))
+            }
+            if viewModel?.isProteusVerified ?? false {
+                statusStackView.addArrangedSubview(UIImageView(image: verifiedImage))
+            }
+            updateLabel()
+            setupAccessibility()
+        }
+    }
 
-        setUpDeviceIconView()
+    private let verifiedImage = Asset.Images.verifiedShield.image.resizableImage(withCapInsets: .zero)
+    private var mlsInfoHeighConstraint: NSLayoutConstraint { mlsThumbprintLabel.heightAnchor.constraint(equalToConstant: 0)
+    }
 
-        deviceTypeIconView.translatesAutoresizingMaskIntoConstraints = false
-        deviceTypeIconView.contentMode = .center
+    // MARK: - Initialization
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        createConstraints()
+        setupStyle()
+    }
 
-        verifiedIconView.image = WireStyleKit.imageOfShieldverified
-        verifiedIconView.translatesAutoresizingMaskIntoConstraints = false
-        verifiedIconView.contentMode = .center
-        verifiedIconView.isAccessibilityElement = true
-        verifiedIconView.accessibilityIdentifier = "device_cell.verifiedShield"
+    @available(*, unavailable)
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
 
+    // MARK: - Methods
+    func setupStyle() {
+        nameLabel.accessibilityIdentifier = "device name"
+        labelLabel.accessibilityIdentifier = "device label"
+        proteusIdLabel.accessibilityIdentifier = "device proteus ID"
+        mlsThumbprintLabel.accessibilityIdentifier = "device mls thumbprint"
```

Comment:
**suggestion**: Changing or adding any accessibilityIdentifiers will impact and may disrupt QA during their own testing. Can you please inform QA that those changes are to happen so they can update their code too?

Reply:
Noted